marklogic-client-api/src/test/java/com/marklogic/client/test/rows/RowManagerValidateDocTest.java

```diff
@@ -0,0 +1,189 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.client.test.rows;
+
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.WriteBatcher;
+import com.marklogic.client.document.DocumentManager;
+import com.marklogic.client.document.XMLDocumentManager;
+import com.marklogic.client.expression.PlanBuilder;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.io.marker.AbstractWriteHandle;
+import com.marklogic.client.row.RowRecord;
+import com.marklogic.client.test.Common;
+import com.marklogic.client.type.PlanParamExpr;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.marklogic.client.io.Format.JSON;
+import static com.marklogic.client.io.Format.XML;
+import static org.junit.Assert.*;
+
+public class RowManagerValidateDocTest extends AbstractOpticUpdateTest {
+
+    private Set<String> expectedUris = new HashSet<>();
+    private DataMovementManager dataMovementManager;
+
+    @Before
+    public void moreSetup(){
+        dataMovementManager = Common.client.newDataMovementManager();
+    }
+// TODO : test with rest-writer user when https://bugtrack.marklogic.com/57978 is fixed
+    @Test
+    public void xmlSchema() {
+        expectedUris.clear();
+        WriteBatcher writeBatcher = dataMovementManager.newWriteBatcher();
+        DocumentMetadataHandle meta = newDefaultMetadata();
+        dataMovementManager.startJob(writeBatcher);
+        final int uriCountToWrite = 10;
+        for (int i = 0; i < uriCountToWrite; i++) {
+            String uri = "/acme/" + i + ".xml";
+            writeBatcher.addAs(uri, meta, new StringHandle("<Doc><key>" + i + "</key><Value>value" + i + "</Value></Doc>").withFormat(XML));
+            expectedUris.add(uri);
+        }
+        writeBatcher.flushAndWait();
+        dataMovementManager.stopJob(writeBatcher);
+
+        PlanBuilder.Plan plan = op
+            .fromDocUris(op.cts.directoryQuery("/acme/"))
+            .joinDoc(op.col("doc"), op.col("uri"))
+            .validateDoc(op.col("doc"),
+                op.schemaDefinition("xmlSchema").withMode("lax"),
+                op.errorDispositionDef(100, "summary")
+            );
+        XMLDocumentManager mgr = Common.client.newXMLDocumentManager();
+        expectedUris.forEach(uri -> assertNotNull("URI was not written: " + uri, mgr.exists(uri)));
+
+        List<RowRecord> rows = resultRows(plan);
+
+        //TODO : uncomment lines 84 and 89 after https://bugtrack.marklogic.com/57987 is fixed
+        List<String> persistedUris = rows.stream().map(row -> {
```

Comment:
Note that this won't be an issue in the test if you're only writing 2 docs, as you'll get only 1 row back (for the valid doc) and can safely verify its contents. Though, it's more important to verify that the doc was written.